### ISSUE TITLE
Help on building and running the examples... does not build

### ISSUE THREAD

**Issue by valioiv** [danielzfranklin/evdi-rs#15]
I want to build and run the examples but I'm getting that error, so I'm not sure how to proceed. Can you please give me some directions?

![image](https://user-images.githubusercontent.com/25353306/220969544-f9b1cbe6-01ec-4aa8-9177-14343b08b19f.png)

**Comment by valioiv** [comment]
I've just used **+nightly** rust and managed to overcome this. But I end up to another issue... @danielzfranklin  , can you please help on this? The best would be to provide up-to-date instruction on building and running all the examples...

![Screenshot from 2023-02-24 10-22-07](https://user-images.githubusercontent.com/25353306/221128874-6e0de7f7-27d0-4be4-b4cf-8414a8d31377.png)

**Comment by dzfranklin** [comment]
This crate is alpha quality and incomplete, but I'd only noted that in the docs and not the readme (just fixed that). I have no idea if it will work for you, and I have no plans to spend a significant time working on it. I built this for a project I gave up on two years ago.

I just fixed those build issues so it now builds on stable.

In terms of examples you'll want to look at https://docs.rs/evdi/latest/evdi/#basic-usage. You'll need the library this wraps (https://github.com/DisplayLink/evdi) to be set up on your compute for that to work. You'll have to look through their documentation for how to do that, I can't remember. I think on ubuntu you just need to `apt install evdi-dkms libevdi0 libevdi0-dev`. After that you'll need to add at least one evdi device, which `bin/evdi_device_add.rs` does.

Feel free to ask more questions if you run into specific issues, I'll see if I can remember.

**Comment by valioiv** [comment]
@danielzfranklin , thanks a lot for your help! It definitely compiles and runs now! I'll further debug it but probably it is some DRM system issue on my setup, not related to **evdi-rs**:

![Screenshot from 2023-03-07 16-49-37](https://user-images.githubusercontent.com/25353306/223459213-3eb399a1-ff51-48f6-9fec-6d22cc72a917.png)

But I do see some **EVDI** activity initiated by the **profile_basic_usage library**!

In general I'm kind of familiar with **EVDI** and I managed to make it running from some C examples and it definitely works. But I'm pretty new in Rust, so I needed some interesting Rust library to start playing with in a real project, so **evdi-rs** catched my eye! Thanks for now and all the best!

**Comment by dzfranklin** [comment]
@valioiv Glad to hear it.

The invalid argument log looks significant to me. I would try commenting out code in the example to see what call is generating that. (Unfortunately evdi logs don't come with a level, the info is an arbitrary level I picked to show them all at)

Let me point out that https://github.com/danielzfranklin/evdi-sys/ is the actual bindings generated from the headers.

I abandoned this library before I used it for anything significant, so keep in mind it's not great rust code.
